src/Data/List/Relation/Unary/All/Properties.agda

```diff
@@ -533,6 +549,16 @@ module _ {P : A → Set p} {Q : A → Set q} (P? : Decidable P) where
   filter⁻ {x ∷ xs}       all⁺  (qx ∷ all⁻) | no    _ | yes  ¬Px = qx ∷ filter⁻ all⁺ all⁻
   filter⁻ {x ∷ xs}       all⁺        all⁻  | no  ¬Px | no  ¬¬Px = contradiction ¬Px ¬¬Px
 
+------------------------------------------------------------------------
+-- partition
+
+module _ {P : A → Set p} (P? : Decidable P) where
+
+  partition-All : ∀ xs → (let ys , zs = partition P? xs) →
+                  All P ys × All (∁ P) zs
+  partition-All xs rewrite partition-defn P? xs =
+    all-filter P? xs , all-filter (∁? P?) xs
```

Comment:
This does two full passes. Yes, it's going to be a lot easier to reason about. It still feels... wrong.